client-sdk/ts-web/rt/src/types.ts

```diff
@@ -18,6 +18,42 @@ export interface CoreCallDataPublicKeyQueryResponse {
     public_key: KeyManagerSignedPublicKey;
 }
 
+/**
+ * Response to the RuntimeInfo query.
+ */
+export interface CoreRuntimeInfoQueryResponse {
+    runtime_version: Version;
+    state_version: number;
+    modules: {[key: string]: ModuleInfo};
+}
+
+export interface Version {
+    major: number;
+    minor: number;
+    patch: number;
+}
+
+/**
+ * Metadata for an individual module within the runtime.
+ */
+export interface ModuleInfo {
+    version: number;
+    params: any;
+    methods: MethodHandlerInfo[];
+}
+
+export interface MethodHandlerInfo {
+    name: string;
+    kind:
+        | typeof METHODHANDLERKIND_CALL
+        | typeof METHODHANDLERKIND_QUERY
+        | typeof METHODHANDLERKIND_PREFETCH;
+}
+
+export const METHODHANDLERKIND_CALL = 'Call';
```

Comment:
I've been putting the constants in a typescript module roughly matching the runtime module name--`core.ts` in this case. But then we wouldn't be able to do the `typeof METHODHANDLERKIND_CALL` etc above due to the way the dependencies flow. The sad thing is that we haven't been listing the variants in the type definition in these actually-enum-typed places.

Would it be okay to hardcode the values in the type definition above?

Reply:
I don't mind if you don't :). I hardcoded the values and left a comment to help keep the two sets of strings in sync.

Reply:
:+1: